core/src/util/vboMesh.cpp

```diff
@@ -67,34 +67,42 @@ void VboMesh::update(GLintptr _offset, GLsizei _size, unsigned char* _data) {
     m_dirty = true;
 }
 
-void VboMesh::subDataUpload() {
-    if (m_dirtySize != 0) {
-        glBindBuffer(GL_ARRAY_BUFFER, m_glVertexBuffer);
-
-        size_t vertexBytes = m_nVertices * m_vertexLayout->getStride();
+bool VboMesh::subDataUpload() {
+    if (m_dirtySize == 0) {
+        return false;
+    }
+    
+    glBindBuffer(GL_ARRAY_BUFFER, m_glVertexBuffer);
 
-        // updating the entire buffer
-        if (vertexBytes - m_dirtySize < m_vertexLayout->getStride()) {
+    long vertexBytes = m_nVertices * m_vertexLayout->getStride();
 
-            // invalidate the data store on the driver
-            glBufferData(GL_ARRAY_BUFFER, vertexBytes, NULL, m_hint);
+    // updating the entire buffer
+    if (m_dirtySize - vertexBytes < m_vertexLayout->getStride()) {
```

Comment:
Oh yes! This was making sense with  std::abs, not anymore.

On Tue, Jul 7, 2015 at 7:30 PM, Matt Blair <notifications@github.com>
wrote:

>>  
>> -            // invalidate the data store on the driver
>> -            glBufferData(GL_ARRAY_BUFFER, vertexBytes, NULL, m_hint);
>> +    // updating the entire buffer
>> +    if (m_dirtySize - vertexBytes < m_vertexLayout->getStride()) {
> Maybe I'm confused, but shouldn't this be `vertexBytes - m_dirtySize` ? `vertexBytes` should always be larger than `m_dirtySize` so this comparison doesn't make sense to me. 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/tangrams/tangram-es/pull/176/files#r34103294